simple_message/CMakeLists.txt

```diff
@@ -90,56 +87,58 @@ add_library(${PROJECT_NAME}_dummy ${CMAKE_CURRENT_BINARY_DIR}/${PROJECT_NAME}_du
-target_link_libraries(simple_message_bswap ${catkin_LIBRARIES})
-add_dependencies(simple_message_bswap ${industrial_msgs_EXPORTED_TARGETS})
+add_library(${PROJECT_NAME}_bswap ${SRC_FILES})
+set_target_properties(${PROJECT_NAME}_bswap PROPERTIES COMPILE_DEFINITIONS "BYTE_SWAPPING")
+target_link_libraries(${PROJECT_NAME}_bswap ${catkin_LIBRARIES})
+add_dependencies(${PROJECT_NAME}_bswap ${industrial_msgs_EXPORTED_TARGETS})
 
 # ALTERNATIVE LIBRARY (64-bit floats)
-add_library(simple_message_float64 ${SRC_FILES})
-set_target_properties(simple_message_float64 PROPERTIES COMPILE_DEFINITIONS "FLOAT64")
-target_link_libraries(simple_message_float64 ${catkin_LIBRARIES})
-add_dependencies(simple_message_float64 ${industrial_msgs_EXPORTED_TARGETS})
+add_library(${PROJECT_NAME}_float64 ${SRC_FILES})
+set_target_properties(${PROJECT_NAME}_float64 PROPERTIES COMPILE_DEFINITIONS "FLOAT64")
+target_link_libraries(${PROJECT_NAME}_float64 ${catkin_LIBRARIES})
+add_dependencies(${PROJECT_NAME}_float64 ${industrial_msgs_EXPORTED_TARGETS})
 
 if(CATKIN_ENABLE_TESTING)
 
     catkin_add_gtest(utest ${UTEST_SRC_FILES})
     set_target_properties(utest PROPERTIES COMPILE_DEFINITIONS "TEST_PORT_BASE=11000")
-    target_link_libraries(utest simple_message)
+    target_link_libraries(utest ${PROJECT_NAME})
 
     catkin_add_gtest(utest_byte_swapping ${UTEST_SRC_FILES})
     set_target_properties(utest_byte_swapping PROPERTIES COMPILE_DEFINITIONS "TEST_PORT_BASE=12000")
-    target_link_libraries(utest_byte_swapping simple_message_bswap)
+    target_link_libraries(utest_byte_swapping ${PROJECT_NAME}_bswap)
 
     catkin_add_gtest(utest_float64 ${UTEST_SRC_FILES})
     set_target_properties(utest_float64 PROPERTIES COMPILE_DEFINITIONS "TEST_PORT_BASE=13000;FLOAT64")
-    target_link_libraries(utest_float64 simple_message_float64)
+    target_link_libraries(utest_float64 ${PROJECT_NAME}_float64)
 
     catkin_add_gtest(utest_udp ${UTEST_SRC_FILES})
     set_target_properties(utest_udp PROPERTIES COMPILE_DEFINITIONS "TEST_PORT_BASE=15000;UDP_TEST")
-    target_link_libraries(utest_udp simple_message)
+    target_link_libraries(utest_udp ${PROJECT_NAME})
 
 endif()
 
+install(TARGETS ${PROJECT_NAME}_dummy
+ DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION})
+        
 install(
-    TARGETS simple_message simple_message_bswap simple_message_float64 
+    TARGETS ${PROJECT_NAME} ${PROJECT_NAME}_bswap ${PROJECT_NAME}_float64 
     ARCHIVE DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
     LIBRARY DESTINATION ${CATKIN_PACKAGE_LIB_DESTINATION}
-    RUNTIME DESTINATION ${CATKIN_GLOBAL_BIN_DESTINATION})
+    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
 
 install(
-    DIRECTORY include/${PROJECT_NAME}/
-    DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION})
+    DIRECTORY include/${PROJECT_NAME}
+    DESTINATION ${CATKIN_PACKAGE_INCLUDE_DESTINATION}
+    RUNTIME DESTINATION ${CATKIN_PACKAGE_BIN_DESTINATION})
```

Comment:
Travis is failing on these lines. Are they necessary for `install(..)` commands that install headers? I was under the impression only libraries needed 'special care' like this (due to Windows using different paths for them, etc).

Reply:
I confirm it's not necessary!

Reply:
>  I was under the impression only libraries needed 'special care' like this (due to Windows using different paths for them, etc).

I think this was fixed at some point, http://docs.ros.org/melodic/api/catkin/html/howto/format2/building_libraries.html#installing should be fine

Reply:
`RUNTIME DESTINATION` is only valid for target-based install rules, please don't specify it for include directories